packages/dialect-react-sdk/hooks/useAddresses.ts

```diff
@@ -0,0 +1,398 @@
+import type {
+  Address,
+  DappAddress,
+  AddressType,
+  DialectSdkError,
+} from '@dialectlabs/sdk';
+import { useCallback, useEffect, useMemo, useState } from 'react';
+import useSWR from 'swr';
+import { EMPTY_OBJ } from '../utils';
+import {
+  WALLET_ADDRESSES_CACHE_KEY_FN,
+  WALLET_DAPP_ADDRESSES_CACHE_KEY_FN,
+} from './internal/swrCache';
+import useDialectDapp from './useDialectDapp';
+import useDialectSdk from './useDialectSdk';
+import useDialectWallet from './useDialectWallet';
+
+interface CreateUpdateParams {
+  type: AddressType;
+  value: string;
+}
+
+interface DeleteParams {
+  type: AddressType;
+}
+
+interface ToggleParams {
+  type: AddressType;
+  enabled: boolean;
+}
+
+interface VerifyParams {
+  type: AddressType;
+  code: string;
+}
+interface ResendParams {
+  type: AddressType;
+}
+
+type AddressActionParams =
+  | CreateUpdateParams
+  | DeleteParams
+  | ToggleParams
+  | VerifyParams
+  | ResendParams;
+
+type AddressEnriched = Address & {
+  enabled: boolean;
+  dappAddress?: DappAddress;
+};
+interface UseAddressesValue {
+  addresses: Record<AddressType, AddressEnriched> | Record<string, never>;
```

Comment:
why type like this?

Reply:
My main concern is about `Record<string, never>`

Reply:
@zhelezkov I added it to support empty object as value — https://stackoverflow.com/a/68208381/10197410, but prob it's not necesary